lib/ansible/parsing/vault/__init__.py

```diff
@@ -690,32 +704,39 @@ def encrypt(self, plaintext, password):
     def decrypt(self, message, password, version=None):
         message = message.replace(b'\n', '')
 
-        # SPLIT SALT, DIGEST, AND DATA
-        data = unhexlify(message)
-        salt, mac, cryptedData = data.split(b"\n", 2)
-        salt = unhexlify(salt)
-        ciphertext = unhexlify(cryptedData)
-
-        key1, key2, iv = self.gen_key_initctr(password, salt)
+        if version == b'1.1':
+            message = unhexlify(message)
+            salt, mac, ciphertext = message.split(b'\n', 2)
+            salt = unhexlify(salt)
+            mac = unhexlify(mac)
+            ciphertext = unhexlify(ciphertext)
+            key1, key2, iv = self.gen_key_initctr(password, salt, generate_iv=True)
+        else:
+            message = base64.b64decode(message)
+            salt = message[0:32]
+            mac = message[32:64]
+            ciphertext = message[64:]
+            key1, key2, iv = self.gen_key_initctr(password, salt)
 
         # EXIT EARLY IF MAC DOESN'T VALIDATE
         hmac = HMAC.new(key2, ciphertext, SHA256)
-        if not self.is_equal(mac, to_bytes(hmac.hexdigest())):
+        if not self.is_equal(mac, to_bytes(hmac.digest())):
             return None
 
         # SET THE COUNTER AND THE CIPHER
-        ctr = Counter.new(128, initial_value=int(iv, 16))
+        ctr = Counter.new(128, initial_value=iv)
```

Comment:
If you're willing to modify the code further, then yes
 1. using salt the size of SHA-256 output for PBKDF2 
 2. using the initial counter value (nonce) directly from os.urandom

Would be the best possible approach. (The first is a *bit* of overkill, but since we're using AES-256, it's probably safer this way)

Ah, and by the way. I think it would be good to add comments like "We're using SHA-256 because we're using AES with 256-bit key size", "There's no need to use bigger salt size than 32 bytes as we're using PBKDF2 with SHA-256, which has 32 byte output" and "The maximum size of the nonce is the block size of cipher, 16 bytes in case of AES" to explain to people reading the code later why we're using such and such magic values and not others (and as a hint when they may need to be changed).